src/core/server/elasticsearch/scoped_cluster_client.test.ts

```diff
@@ -134,6 +146,33 @@ describe('#callAsCurrentUser', () => {
     expect(internalAPICaller).not.toHaveBeenCalled();
   });
 
+  test('callAsCurrentUser allows passing additional headers', async () => {
+    const mockResponse = { data: 'response' };
+    scopedAPICaller.mockResolvedValue(mockResponse);
+    await expect(
+      clusterClient.callAsCurrentUser('security.authenticate', {
+        some: 'some',
+        headers: { additionalHeader: 'Oh Yes!' },
+      })
+    ).resolves.toBe(mockResponse);
+    expect(scopedAPICaller).toHaveBeenCalledTimes(1);
+    expect(scopedAPICaller).toHaveBeenCalledWith(
+      'security.authenticate',
+      { some: 'some', headers: { one: '1', additionalHeader: 'Oh Yes!' } },
+      undefined
+    );
+    scopedAPICaller.mockClear();
```

Comment:
mock won't be cleared up if assertion above throws

Reply:
good catch. Added it to the top of cases instead of bottom :+1: